tests/test_hypothesis.py

```diff
@@ -0,0 +1,186 @@
+            "expected_result": ("2020-01-01", "2020-01-06"),
+        }
+        return data
+
+    @pytest.fixture
+    def mock_df(self):
+        return pd.DataFrame(
+            {
+                "Version": [2],
+                "Date_of_report": ["2022-01-03T09:00:00Z"],
+                "Date_of_statistics": ["2022-01-01"],
+                "Security_region_code": ["VR01"],
+                "Security_region_name": ["Groningen"],
+                "Tested_with_result": [976],
+                "Tested_positive": [378],
+            }
+        )
+
+    @pytest.fixture
+    def file_list(self):
+        return [
+            "data-rivm/tests/rivm_daily_2021-01-01.csv.gz",
+            "data-rivm/tests/rivm_daily_2021-01-02.csv.gz",
+            "data-rivm/tests/rivm_daily_2021-01-15.csv.gz",
+            "data-rivm/tests/rivm_daily_2021-01-30.csv.gz",
+            "data-rivm/tests/rivm_daily_2021-02-01.csv.gz",
+            "data-rivm/tests/rivm_daily_2021-02-15.csv.gz",
+            "data-rivm/tests/rivm_daily_2021-02-28.csv.gz",
+        ]
+
+    def test_filter_dates_within_range(self, file_list):
+        # Test whether function correctly filter out dates
+        # that are *clearly* within the bounds of `time_period`
+        time_period = ("2021-01-01", "2021-01-02")
+        expected_output = [
+            "data-rivm/tests/rivm_daily_2021-01-01.csv.gz",
+            "data-rivm/tests/rivm_daily_2021-01-02.csv.gz",
+        ]
+        output = Hypothesis.filter_dates(file_list, time_period)
+        assert output == expected_output
+
+    def test_filter_dates_on_boundary(self, file_list):
+        # Test whether function correctly includes
+        # or excludes dates that are exactly on the edge
+        time_period = ("2021-01-16", "2021-02-28")
+        expected_output = [
+            "data-rivm/tests/rivm_daily_2021-01-30.csv.gz",
+            "data-rivm/tests/rivm_daily_2021-02-01.csv.gz",
+            "data-rivm/tests/rivm_daily_2021-02-15.csv.gz",
+            "data-rivm/tests/rivm_daily_2021-02-28.csv.gz",
+        ]
+        output = Hypothesis.filter_dates(file_list, time_period)
+        assert output == expected_output
+
+    def test_range_error(self, file_list):
+        # Test that values outside the bounds raise an error
+        time_period = ("2021-02-01", "2026-03-15")
+        with pytest.raises(
+            ValueError, match=r"time_period .* is outside available dates"
+        ):
+            Hypothesis.filter_dates(file_list, time_period)
+
+    @patch("comma.hypothesis.Hypothesis.get_file_paths")
+    @patch("comma.hypothesis.Hypothesis.filter_dates")
+    @patch("comma.hypothesis.pd.read_csv")
+    # Mock tqdm to just return an iterator
+    @patch("comma.hypothesis.tqdm", side_effect=lambda x, *args, **kwargs: x)
+    def test_get_covid_data(
+        self,
+        mocked_tqdm,
+        mocked_read_csv,
+        mocked_filter_dates,
+        mocked_get_file_paths,
+        mock_df,
+    ):
```

Comment:
This looks a bit like excessive mocking; I would try to limit mocking to functions that depend on external and/or unreliable resources, or are very expensive. Here, why would you not just call `get_file_paths()` etc.?
On the other hand, that would turn this test slightly into an integration test. I think this is a consequence of chaining static methods instead of using object-oriented design, but perhaps that can be resolved on the side of the class rather than in the test design here.